### DISCUSSION TITLE
Automatically compositing images over subimages?

### DISCUSSION THREAD

**Discussion by lunix4all** [ImageMagick/ImageMagick#2346] [Help] [answered]
ImageMagick Version: 6.9.11-24
Platform: Debian Unstable

Hello! I am wondering if it is possible to automatically composite an image over a subimage.

For example, let's say I have multiple images like this:
![image](https://user-images.githubusercontent.com/28959954/88985977-4b184d80-d297-11ea-83ac-577335d5cc46.png)

and I want to composite this image:
![heal2](https://user-images.githubusercontent.com/28959954/89046683-ef8ba580-d312-11ea-8b38-13990717ea76.png)

over this subimage (same size):
![image](https://user-images.githubusercontent.com/28959954/88986242-03de8c80-d298-11ea-8d98-e6a7f0429066.png)

Resulting in this:
![image](https://user-images.githubusercontent.com/28959954/89046879-35e10480-d313-11ea-8beb-403a1f3ca0a1.png)

The position of the subimage can vary across the original images, for
example, this one has it slightly lower:
![image](https://user-images.githubusercontent.com/28959954/88986358-5a4bcb00-d298-11ea-8477-d142e5fbda02.png)

So I would like the program to account for that, so that the dimensions of the image remain unchanged, like this:
![image](https://user-images.githubusercontent.com/28959954/89047222-bdc70e80-d313-11ea-9bb6-8b1ef26179df.png)

Is there any way to do this?

**Comment by snibgo** [comment, marked as answer]
Find the position of the subimage (I call this carSub.png) in the main image (car1.png):
```
magick compare -metric RMSE car1.png carSub.png  -subimage-search NULL:
0 (0) @ 155,163
```

Use the found coordinates to composite the required replacement (carSubRep.png):
```
magick car1.png carSubRep.png -geometry +155+163 -composite out.png
```

This doesn't work on your second example (car2.png) because your carSub.png has pixels that are not in car2.png. You could trim a few rows from the bottom of carSub.png first.

**Reply by lunix4all** [reply]
How would you automate this process for a large set of images?

**Reply by fmw42** [reply]
Write a shell script loop over each image pair.

**Comment by snibgo** [comment]
Call that script inside a shell loop. For example, see https://github.com/ImageMagick/ImageMagick/discussions/2344

**Comment by lunix4all** [comment]
Would this script do what I am looking for?

https://www.imagemagick.org/Usage/scripts/overlap

**Reply by fmw42** [reply]
I do not know. The comments seems to imply some similarity, but it seems overkill for what you want to do. You can just try it to see what happens. Otherwise, just write a script loop that does the two commands that snibgo suggested.
